xml/hwbp_registers.xml

```diff
@@ -777,18 +777,30 @@
     <register name="Instruction Count" short="icount" address="0x7a1">
         This register is accessible as \RcsrTdataOne when \FcsrTdataOneType is 3.
 
-        If \FcsrIcountCount is not 0, then every instruction completed or
+        When \FcsrIcountCount is greater than 1, every instruction completed or
         trap taken from a mode where the trigger is enabled decrements
         \FcsrIcountCount by 1.
-        When
-        \FcsrIcountCount is decremented from 1 to 0 then \FcsrIcountPending
-        becomes set. When \FcsrIcountPending is set, the trigger fires just
+
+        When \FcsrIcountCount is 1, and an instruction is completed or trap
+        taken from a mode where the trigger is enabled, then \FcsrIcountPending
+        becomes set. In addition \FcsrIcountCount may become 0 as well.
+
+        The only exception to the above is when the instruction executed is a
+        write to the icount trigger. In that case \FcsrIcountPending may or may
+        not become set if \FcsrIcountCount was 1. Afterwards \FcsrIcountCount
+        contains the newly written value.
+
+        When \FcsrIcountCount is 0 it stays at 0.
+
+        When \FcsrIcountPending is set, the trigger fires just
         before any further instructions are executed in a mode where the
         trigger is enabled. As the trigger fires, \FcsrIcountPending is
+        cleared. In addition, if \FcsrIcountCount is 1 then \FcsrIcountM,
+        \FcsrIcountS, \FcsrIcountU, \FcsrIcountVs, and \FcsrIcountVu are all
```

Comment:
This isn't currently required, though the count field says that this is allowed.  For instance, if icount.m=s=0 and icount.u=1 then there's no need to clear icount.u when trapping to M or S.  And there's no requirement to clear any of the mode filters on an action=1 trigger.  So this should change to be an allowed but not required behavior.

Reply:
I think this is right. If icount.count=1 when pending is set, that means count was hardwired to 1. Otherwise it would be 0. But I'll clarify that this only applies when count is hardwired to 1.

Reply:
> If icount.count=1 when pending is set, that means count was hardwired to 1. Otherwise it would be 0.

Or it means that software wrote pending=1, count=0.  Useful?  No.  Possible?  Yes.